### PR TITLE
enhancement: Add ability to attach files to email

### PR DESCRIPTION
**Pull request by kaymckay**
PR for https://github.com/go-vela/community/issues/513

adds ability to attach files to an email. This PR corrects the use of email.Attachment and adds EmailFilename to be the file that contains the contents of the email.

### CODE REVIEW

File: cmd/vela-email/plugin.go [hunk 1]
```diff
@@ -118,6 +120,22 @@ func (p *Plugin) Validate() error {
 		return ErrorMissingEmailFromParam
 	}
 
+	if len(p.Attachment.Filename) != 0 {
+		fileInfo, err := os.Stat(p.Attachment.Filename)
+		if errors.Is(err, os.ErrNotExist) {
```
Comment (wass3r) [inline cmd/vela-email/plugin.go:125]: are there other errors we should worry about?

File: cmd/vela-email/plugin.go [hunk 2]
```diff
@@ -77,17 +79,17 @@ func (p *Plugin) Validate() error {
 	defer logrus.Trace("exited plugin.Validate")
 
 	logrus.Info("Validating Parameters...")
-	if len(p.Attachment.Filename) != 0 {
-		fileInfo, err := os.Stat(p.Attachment.Filename)
+	if len(p.EmailFilename) != 0 {
+		fileInfo, err := os.Stat(p.EmailFilename)
 		if errors.Is(err, os.ErrNotExist) {
```
Comment (wass3r) [inline cmd/vela-email/plugin.go:84]: didn't see this until the next comment i made, but same question regarding other errors.

**Review by wass3r — COMMENTED** [review]
couple of comments below.

also,
1. is this a breaking change?
2. we might need attribution for the test image in use depending on where it came from, unless you created it yourself?

File: cmd/vela-email/plugin.go (hunk 1 above)
Comment (kaymckay) [reply cmd/vela-email/plugin.go:125]: any other errors will get handled by the email library and returned

File: cmd/vela-email/plugin.go (hunk 2 above)
Comment (kaymckay) [reply cmd/vela-email/plugin.go:84]: not that I can think of, like above, any other error should get handled with email.NewEmailFromReader()

**Comment by kaymckay** [comment]
@wass3r  this isn't a breaking change, and i'm not sure what you mean by attribution but i pulled it off the internet. I can replace it with a target image if needed

**Review by wass3r — APPROVED** [review]

**Review by plyr4 — APPROVED** [review]

**Pull request merged**